File: 0695-max-area-of-island/solution.java
class Solution {
    
    public int maxAreaOfIsland(int[][] grid) {
    int max = 0 ;
        for(int i =0 ; i<grid.length; i++)
            for(int j= 0 ; j<grid[i].length ; j++)
                if(grid[i][j]==1)
                    max = Math.max( helper(grid, i, j) ,max);
     return max ;

    }

    private int helper(int[][] grid, int i , int j ){
        if(i<0 || i>=grid.length || j<0 || j>=grid[i].length || grid[i][j]==0) return 0;
        grid[i][j]=0; 
       int area = 1;
        return area +helper(grid, i+1 , j )
        +helper(grid, i-1 , j )
        +helper(grid, i, j+1  )
        +helper(grid, i, j-1 );
        
        

    }
}

File: 0695-max-area-of-island/solution.py
class Solution:
    def maxAreaOfIsland(self, grid: List[List[int]]) -> int:
        def dfs(i, j, grid):
            if i<0 or i>len(grid)-1 or j<0 or j>len(grid[0])-1 or grid[i][j]==0:
                return 0
            else:
                grid[i][j]=0
                return 1+ dfs(i+1,j, grid)+ dfs(i-1,j,grid)+dfs(i,j+1,grid)+dfs(i,j-1,grid)


        area = 0
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j]==1:
                    area = max(dfs(i,j,grid),area)
        
        return area

        
